project-boilerplates/sending-images-streamlit-fastapi/backend/fast_api/api.py

```diff
@@ -0,0 +1,38 @@
+from fastapi import FastAPI, UploadFile, File
+from fastapi.middleware.cors import CORSMiddleware
+
+import numpy as np
+import cv2
+import io
+from starlette.responses import StreamingResponse
+from face_rec.face_detection import annotate_face
+
+app = FastAPI()
+
+# # Allow all requests (optional, good for development purposes)
+# app.add_middleware(
+#     CORSMiddleware,
+#     allow_origins=["*"],  # Allows all origins
+#     allow_credentials=True,
+#     allow_methods=["*"],  # Allows all methods
+#     allow_headers=["*"],  # Allows all headers
+# )
+
+@app.get("/")
+def index():
+    return {"status": "ok"}
+
+@app.post('/upload_image')
+async def receive_image(img: UploadFile=File(...)):
+    ### Receiving and decoding the image
+    contents = await img.read()
```

Comment:
Async/await pattern seems to be [recommended](https://stackoverflow.com/questions/70123888/using-async-def-vs-def-in-fastapi-and-testing-blocking-calls) when endpoints have (potentially heavy, for large images) IO-bound actions. However, in the [FastAPI docs](https://fastapi.tiangolo.com/async/) they also mention that you don't need async/await for CPU-bound operations, which the face detection would be.

But overall, the async/await pattern seems to be prevalent in their docs, especially in endpoints with file uploads, I think it's more stable to avoid data loss during API request.

WDYT?

Reply:
Fair enough! 😉